### PR TITLE
test: drivers: pwm: extends pwm_loopback test

### PR DESCRIPTION
**Pull request by TilmannUnte**
Adds a new zassert to check whether PWM capture has previously been
enabled and correctly returns -EBUSY on a second call.
While working on #39694 this was found to be a missing test case.
The new test case has been verified against the suggested changes of
that PR.

Signed-off-by: Tilmann Unte <unte@es-augsburg.de>

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Thanks, looks good. I am blocking this until #39914 is merged in order not to break testing.

**Review by henrikbrixandersen — APPROVED** [review]
Thanks!

**Comment by henrikbrixandersen** [comment]
@TilmannUnte Please rebase on `main` to fix the compliance issue.

**Pull request merged**
